### PR TITLE
Add a chapter for uninstall package requests in docs

### PR DESCRIPTION
**Pull request by davidcassany**
@schaefi, I have included a new _Gracefully Uninstall System Packages_ chapter in docs. I have some doubts about how this should look like, since it is not exactly a _how to_, there is no step by step recipe, just some explanation over an hypothetical example.

@tomschr, probably you could also give it a quick look, we'd appreciate it. Is a short chapter :wink:

### CODE REVIEW

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+handy under certain circumstances. As an example, someone could be interested
+in performing some configuration tasks in the
+`config.sh` script (see :ref:`prepare step <prepare-step>` for
+further details) that require the inclusion of some extra packages,
+which will be only needed at build time. One example would be compiling
+some unpacked application sources.
+
+KIWI description file schema defines package requests of type `uninstall`
+and type `delete` in order to remove previously installed packages. The
+`uninstall` requests perform a clean packagei removal by removing any
```
Comment (schaefi) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: Typo with packagei

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+package dependent on the requested ones and also removing orphan
+dependencies. The `delete` requests perform a hard removal without any
+dependency check, thus only listed packages are deleted even if it breaks
+dependencies or compromises any underlying package database. This page
+focuses on `uninstall` package requests.
+
+This is an example description of a Container image that removes user related
+tools and development tools:
+
```
Comment (schaefi) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: I very much like the description in text form :+1: 

I think for demo the use case it would be enough to just list the packages sections with type="uninstall" and type="delete". I think all the other parts of the image description are not of any value and might confuse the reader

Comment (schaefi) [reply doc/source/building/working_with_images/uninstall_package_requests.rst]: But I let @tomschr decide if that makes sense or if it's just me :smile:

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   package requests and how it could be used in order to remove
+   packages once the image configuration using the `config.sh`
+   script is done.
+
+Uninstalling packages from the system image that were previously installed
+during the installation phase is an operation that can be
+handy under certain circumstances. As an example, someone could be interested
+in performing some configuration tasks in the
+`config.sh` script (see :ref:`prepare step <prepare-step>` for
+further details) that require the inclusion of some extra packages,
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: I would end the sentence after "futher details)":

```
(see :ref:`prepare step <prepare-step>` for further details). That would require...
```

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+which will be only needed at build time. One example would be compiling
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: Futhermore, I would rephrase the 2nd sentence a bit:

```
That would require to include some extra packages, which are only needed at build time.
```

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+some unpacked application sources.
+
+KIWI description file schema defines package requests of type `uninstall`
+and type `delete` in order to remove previously installed packages. The
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: * Remove "in oder to" (filler words)
* Maybe remove the "to remove previously installed packages" completely? The details are described in the paragraphs below.

Comment (tomschr) [reply doc/source/building/working_with_images/uninstall_package_requests.rst]: Maybe add a linebreak after "packages."?

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+`uninstall` requests perform a clean packages removal by removing any
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: Maybe it would be easier to read if you put these two types into a list:

```
* The `uninstall` requests ...
* The `delete` requests ...
```

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+package dependent on the requested ones and also removing orphan
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: dependent -> dependencies?

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+dependencies. The `delete` requests perform a hard removal without any
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: Are there more than one delete requests? I would use the singular form: "The delete request performs ..."

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+dependency check, thus only listed packages are deleted even if it breaks
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: End the sentence after check:

```
dependency check. Thus, only ...
```

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+      </packages>
+      <!-- These packages will be uninstalled after running config.sh -->
+      <packages type="uninstall">
+        <package name="shadow"/>
+        <package name="make"/>
+        <package name="llvm-clang"/>
+      </packages>
+    </image>
+
+In this example after installing all the packages and archives, image
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: Maybe use the phrase "In the previous example..."

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+repositories are configured and then the `config.sh` script is executed.
+In `config.sh` the `foo_app_sources.tar.gz` could be compiled using the
+`make` and `llvm` packages with something like a `make install` call. It is
+a common practice to build tiny and single purpose container images, thus
+makes sense to remove unneeded packages, like `make` and `llvm-clang`.
+In order to gracefully remove them, they have been included into the
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: Remove "In order to" (filler words) :wink:

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+*type="uninstall"* packages list. Those packages will be removed including
+a dependency cleanup.
+
+.. warning::
+
+   Note that `uninstall` packages requests deletes the listed
+   packages, plus the packages dependent on the listed ones, plus any orphan
+   dependency of the listed packages. Use this feature with caution as it
+   can easily cause the removal of sensitive tools leading to failures in
+   later build stages.
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: Hmn, I have some issues with this warning:

* plural vs. singular. If you stick with plural, it can't be "delete**s**". :wink: 
* it's a bit ambiguous to me. Does it mean to "use" the `uninstall` package requests? Or uninstalling package requests?
* Don't start with "Note" if it's a warning. That's a contradiction. :wink: 

One suggestion:

```
An `uninstall` packages request deletes:

* the listed packages,
* the packages dependent on the listed ones, and
* any orphaned dependecy of the listed packages.

Use this...
```

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+
+In the above example also the *shadow* package is being removed, again, in
+this concrete case, it is not expected to be needed into the final image.
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: * concrete -> specific
* into -> in

File: doc/source/building/working_with_images/uninstall_package_requests.rst [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+The *shadow* package mainly provides tools to handle user accounts and,
+in a container image, once everything is installed and configured, it is
+not expected to require any further user account modification to the image, 
+tools such as *useradd* or *usermod* will not be required.
```
Comment (tomschr) [inline doc/source/building/working_with_images/uninstall_package_requests.rst]: That sentence is too long. Try to shorten it. I would probably end the sentence after "user accounts".

**Comment by davidcassany** [comment]
@tomschr Thanks much :+1:, I am on it

**Comment by davidcassany** [comment]
@tomschr, all covered, if you remove the request for changes and tests are green, we are reday to merge. Thanks!
